calico/felix/plugins/fiptgenerator.py

```diff
@@ -434,31 +434,22 @@ def filter_forward_chain(self, ip_version):
         # the raw table.
         forward_chain.append(
             "--append {chain} --jump ACCEPT --match mark "
-            "--mark {mark}/{mark}".format(
+            "--mark {mark}/{mark} "
+            "--match conntrack --ctstate UNTRACKED".format(
                 chain=CHAIN_FORWARD,
                 mark=self.IPTABLES_MARK_ACCEPT)
         )
 
-        for iface_match in self.IFACE_MATCH:
-            forward_chain.extend(self.drop_rules(
-                ip_version, CHAIN_FORWARD,
-                "--in-interface %s --match conntrack --ctstate "
-                "INVALID" % iface_match, None))
-            forward_chain.extend(
-                self.drop_rules(
-                    ip_version, CHAIN_FORWARD,
-                    "--out-interface %s --match conntrack --ctstate "
-                    "INVALID" % iface_match, None))
-            forward_chain.extend([
-                # First, a pair of conntrack rules, which accept established
-                # flows to/from workload interfaces.
-                "--append %s --in-interface %s --match conntrack "
-                "--ctstate RELATED,ESTABLISHED --jump ACCEPT" %
-                (CHAIN_FORWARD, iface_match),
-                "--append %s --out-interface %s --match conntrack "
-                "--ctstate RELATED,ESTABLISHED --jump ACCEPT" %
-                (CHAIN_FORWARD, iface_match),
-            ])
+        forward_chain.extend(self.drop_rules(
+            ip_version, CHAIN_FORWARD,
+            "--match conntrack --ctstate INVALID", None))
+        forward_chain.extend([
+            # First, a pair of conntrack rules, which accept established
+            # flows to/from workload interfaces.
+            "--append %s --match conntrack "
+            "--ctstate RELATED,ESTABLISHED --jump ACCEPT" %
+            (CHAIN_FORWARD,),
+        ])
```

Comment:
Similarly here, to explain the thinking, which I think is that:

- in principle, the right thing for function is to move these acceptances further down into the per-host-endpoint chains, so that they only apply when host endpoints are in use (and on the relevant reverse flows)

- but we are leaving them here for performance, because moving them down would require a lot more matches before hitting the conntrack rules.